src/commands/price.ts

```diff
@@ -0,0 +1,186 @@
+			() => c("command-option-description").t`Konami's official card database identifier.`
+		);
+		const vendorOption = buildLocalisedCommand(
+			new SlashCommandStringOption().setRequired(true),
+			() => c("command-option").t`vendor`,
+			() => c("command-option-description").t`The vendor to fetch the price data from.`
+		).addChoices(...Object.entries(CHOICES_GLOBAL).map(([value, name]) => buildLocalisedChoice(value, name)));
+		nameSubcommand
+			.addStringOption(nameOption)
+			.addStringOption(vendorOption)
+			.addStringOption(getInputLangStringOption());
+		passwordSubcommand.addStringOption(passwordOption).addStringOption(vendorOption);
+		konamiIdSubcommand.addStringOption(konamiIdOption).addStringOption(vendorOption);
+		builder.addSubcommand(nameSubcommand).addSubcommand(passwordSubcommand).addSubcommand(konamiIdSubcommand);
+		return builder.toJSON();
+	}
+
+	protected override get logger(): Logger {
+		return this.#logger;
+	}
+
+	// TODO: i18n
+	private vendorFormats = {
+		tcgplayer: (price: number): string => `$${price.toFixed(2)}`,
+		cardmarket: (price: number): string => `€${price.toFixed(2)}`,
+		coolstuffinc: (price: number): string => `$${price.toFixed(2)}`
+	};
+
+	async getPrice(card: Static<typeof CardSchema>, vendor: string): Promise<APIPrice | undefined> {
+		if (!card.name.en) {
+			// note: this is about our database, not the user input. Could clarify.
+			// unsure where, how or if to localise this error. Will it even come up?
+			throw new Error(t`Sorry, I can't find the price for a card with no English name!`);
+		}
+		// we make sure the ID for each vendor choice is the same as its required form here
+		const priceUrl = `${process.env.PRICE_API}?cardone=${encodeURIComponent(card.name.en)}&vendor=${vendor}`;
+		const response = await fetch(priceUrl);
+		// 400: Bad syntax, 404: Not found
+		if (response.status === 400 || response.status === 404) {
+			return undefined;
+		}
+		// 200: OK
+		if (response.status === 200) {
+			return await response.json();
+		}
+		throw new Error((await response.json()).message);
+	}
+
+	protected override async execute(interaction: CommandInteraction): Promise<number> {
+		const type = interaction.options.getSubcommand(true) as CardLookupType;
+		const input = interaction.options.getString("input", true);
+		const resultLanguage = await this.locales.get(interaction);
+		const inputLanguage = (interaction.options.getString("input-language") as Locale) ?? resultLanguage;
+		const vendor = interaction.options.getString("vendor", true) as vendorId;
+		// Send out both requests simultaneously
+		const [, card] = await Promise.all([interaction.deferReply(), getCard(type, input, inputLanguage)]);
+		let end: number;
+		if (!card) {
+			end = Date.now();
+			useLocale(resultLanguage);
+			await interaction.editReply({ content: t`Could not find a card matching \`${input}\`!` });
+		} else {
+			const prices = await this.getPrice(card, vendor);
+			end = Date.now();
```

Comment:
There's a lot of work being done after this, so this should be moved further down in the two branches to just before the first editReply each